trunk/src/app/srs_app_http_stream.cpp

```diff
@@ -583,13 +583,13 @@ SrsLiveStream::SrsLiveStream(SrsRequest* r, SrsBufferCache* c)
     cache = c;
     req = r->copy()->as_http();
     security_ = new SrsSecurity();
-    alive_viewers_ = 0;
 }
 
 SrsLiveStream::~SrsLiveStream()
 {
     srs_freep(req);
     srs_freep(security_);
+    viewers_.clear();
```

Comment:
viewers need to be empty here, otherwise the loop in `do_serve_http` will break, and `server_http` will continue run. 

So add `srs_assert(viewers_.size() == 0);`, before `viewers.clear()`.

Reply:
Fixed.